### PR TITLE
=clu #16667 fix Inconsistent message order in ClusterMessageSerializer

### PR DESCRIPTION
**Pull request by He-Pin**
add the lost `toVector` call

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala [hunk 1]
```diff
@@ -309,9 +307,9 @@ class ClusterMessageSerializer(val system: ExtendedActorSystem) extends Serializ
 
   private def metricsGossipEnvelopeToProto(envelope: MetricsGossipEnvelope): cm.MetricsGossipEnvelope = {
     val mgossip = envelope.gossip
-    val allAddresses = mgossip.nodes.foldLeft(Set.empty[Address])((s, n) ⇒ s + n.address)
+    val allAddresses: Vector[Address] = mgossip.nodes.foldLeft(Set.empty[Address])((s, n) ⇒ s + n.address).toVector
```
Comment (hepin1989) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: this line the the actual fix

File: akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala [hunk 2]
```diff
@@ -309,9 +307,9 @@ class ClusterMessageSerializer(val system: ExtendedActorSystem) extends Serializ
...
     val addressMapping = allAddresses.zipWithIndex.toMap
-    val allMetricNames = mgossip.nodes.foldLeft(Set.empty[String])((s, n) ⇒ s ++ n.metrics.iterator.map(_.name))
+    val allMetricNames: Vector[String] = mgossip.nodes.foldLeft(Set.empty[String])((s, n) ⇒ s ++ n.metrics.iterator.map(_.name)).toVector
```
Comment (hepin1989) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: this one too

**Comment by drewhk** [comment]
OK TO TEST

File: akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: the `foldLeft` can be replaced by `map`

Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: and possibly with `(breakOut)`

**Comment by patriknw** [comment]
awesome finding @hepin1989 , I can understand that it was difficult to track it down

File: akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala (hunk 1 above)
Comment (hepin1989) [inline akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala]: the minimal change show the  issue straightly.
will update later

**Comment by hepin1989** [comment]
@patriknw updated and rebased

**Comment by patriknw** [comment]
LGTM, thanks

**Comment by rkuhn** [comment]
LGTM; @patriknw do you want to do the honors?

**Comment by patriknw** [comment]
Refs #16667

**Comment by patriknw** [comment]
Thank you @hepin1989

**Comment by patriknw** [comment]
@hepin1989 will you do the forward port to master (including same fix in new metrics module)?

**Pull request merged**